### PR TITLE
Get rid of notification center and replace it with support for multiple access requests

### PR DESCRIPTION
**Pull request by KillariDev**
fix https://github.com/DarkFlorist/TheInterceptor/issues/490

image

### CODE REVIEW

**Review by MicahZoltu — APPROVED** [review]

File: app/ts/background/windows/interceptorAccess.ts [hunk 1]
```diff
@@ -114,212 +99,178 @@ export async function requestAccessFromUser(
...
-	if (pendingInterceptorAccess !== undefined) return rejectReply()
-
+	let justAddToPending = false
+	if (pendingInterceptorAccess.size !== 0) justAddToPending = true
+	
 	// check if we need to ask address access or not. If address is put to never need to have address specific permision, we don't need to ask for it
 	const askForAddressAccess = requestAccessToAddress !== undefined && settings.userAddressBook.addressInfos.find((x) => x.address === requestAccessToAddress.address)?.askForAddressAccess !== false
 	const accessAddress = askForAddressAccess ? requestAccessToAddress : undefined
+	const future = new Future<InterceptorAccessReply>()
+	const requestId = request === undefined ? -Math.random() : request.requestId // if there's no particular request requesting this access, generate random ID for it
```
Comment (MicahZoltu) [inline app/ts/background/windows/interceptorAccess.ts:111]: Why not just a file-local constant that increments over time?  Does this need to be unique across runs or something?

**Review by MicahZoltu — APPROVED** [review]
This is a pretty big change, would be good to get more eyes on it.

File: app/ts/background/windows/interceptorAccess.ts (hunk 1 above)
Comment (KillariDev) [reply app/ts/background/windows/interceptorAccess.ts]: I'll realized it has to be consistent between sessions too. I'll keep it as random

**Comment by KillariDev** [comment]
I refactored pr quite a bit and this should also be solved https://github.com/DarkFlorist/TheInterceptor/issues/480

File: app/ts/background/windows/interceptorAccess.ts (hunk 1 above)
Comment (KillariDev) [reply app/ts/background/windows/interceptorAccess.ts]: I also realized I can generate that Id deterministically :)

File: app/ts/components/pages/InterceptorAccess.tsx [hunk 2]
```diff
@@ -1,20 +1,58 @@
...
+		}) }
+	</div>
+}
+
 function AssociatedTogether({ associatedAddresses, renameAddressCallBack }: { associatedAddresses: readonly AddressInfoEntry[], renameAddressCallBack: RenameAddressCallBack } ) {
 	const [showLogs, setShowLogs] = useState<boolean>(associatedAddresses.length > 1)
 
 	return <>
-		<div class = 'card' style = 'margin-top: 10px; margin-bottom: 10px'>
+		<div class = 'card' style = 'margin-top: 10px; margin-bottom: 10px;'>
```
Comment (jubalm) [inline app/ts/components/pages/InterceptorAccess.tsx:55]: there are `<p>` tags inside `<p>` elements within this code block which could potentially display incorrectly. consider using `<span>` to be safe

File: app/ts/components/pages/InterceptorAccess.tsx [hunk 3]
```diff
@@ -1,20 +1,58 @@
...
+	</header>
+}
+
+type UnderTransactionsParams = {
+	reversedPendingAccessRequestArray: PendingAccessRequestArray
+}
+
+function UnderAccesses(param: UnderTransactionsParams) {
+	const nTx = param.reversedPendingAccessRequestArray.length
+	return <div style = {`position: relative; top: ${ nTx * -HALF_HEADER_HEIGHT }px;`}>
```
Comment (jubalm) [inline app/ts/components/pages/InterceptorAccess.tsx:40]: consider using objects as style properties here
```suggestion
	return <div style = {{ position: 'relative', top: nTx * -HALF_HEADER_HEIGHT }}>
```

**Review by jubalm — COMMENTED** [review]
Consider using object notation for inline styles. As a bonus, TypeScript can help ensure spelling and add unit `px` to values.

```ts
// string
<p style = `fon-size: 13px; margin-bottom: ${SOMEREFERENCE / 2}px`>

// object notation
<p style = {{ fontSize: 13, marginBottom: SOMEREFERENCE / 2 }}>
```

File: app/ts/components/pages/InterceptorAccess.tsx (hunk 2 above)
Comment (KillariDev) [reply app/ts/components/pages/InterceptorAccess.tsx:55]: I created issue about this. I didn't know that its not allowed

File: app/ts/components/pages/InterceptorAccess.tsx (hunk 3 above)
Comment (KillariDev) [reply app/ts/components/pages/InterceptorAccess.tsx:40]: added issue about this, I have been planning to move into this styling but I havent gotten aroudn it

**Pull request merged**